SYCL/SubGroup/generic-shuffle.cpp

```diff
@@ -225,11 +225,14 @@ int main() {
   check_struct<class KernelName_zHfIPOLOFsXiZiCvG, std::complex<float>>(
       Queue, ComplexFloatGenerator);
 
-  auto ComplexDoubleGenerator = [state = std::complex<double>(0, 1)]() mutable {
-    return state += std::complex<double>(2, 2);
-  };
-  check_struct<class KernelName_CjlHUmnuxWtyejZFD, std::complex<double>>(
-      Queue, ComplexDoubleGenerator);
+  if (Queue.get_device().has(sycl::aspect::fp64)) {
+    auto ComplexDoubleGenerator = [state =
+                                       std::complex<double>(0, 1)]() mutable {
+      return state += std::complex<double>(2, 2);
+    };
+    check_struct<class KernelName_CjlHUmnuxWtyejZFD, std::complex<double>>(
+        Queue, ComplexDoubleGenerator);
+  }
 
   std::cout << "Test passed." << std::endl;
```

Comment:
Should this test print something like "Some tests skipped because device doesn't support aspect::fp64."?

I'm asking because I noticed that broadcast_fp16.cpp prints something like this.  I don't know what the convention is.

Reply:
broadcast_fp16.cpp prints it because the entire test requires fp16. When it's only parts of the test that are masked out we don't usually print anything, but I can add it no problem.